src/Common/Abstracts/Custom_List_Table.php

```diff
@@ -0,0 +1,184 @@
+		$repo->set_use_default_args( true );
+
+		return $result;
+	}
+
+	/**
+	 * Returns the repository.
+	 *
+	 * @since TBD
+	 *
+	 * @return Repository The repository.
+	 */
+	abstract protected function get_repository(): Repository;
+
+	/**
+	 * Returns the items for the current page.
+	 *
+	 * @since TBD
+	 *
+	 * @param int $per_page The number of items to display per page.
+	 *
+	 * @return array The items for the current page.
+	 */
+	protected function get_items( int $per_page ): array {
+		$repo = $this->get_repository();
+
+		$repo->set_use_default_args( false );
+		$results = $repo->by_args( $this->get_arguments() )->page( $this->get_pagenum() )->per_page( $per_page )->all();
+		$repo->set_use_default_args( true );
+
+		return $results;
+	}
+
+	/**
+	 * Returns whether the list is completely empty.
+	 *
+	 * @since TBD
+	 *
+	 * @return bool
+	 */
+	public function is_empty(): bool {
+		$repo = $this->get_repository();
+		$repo->set_use_default_args( false );
+		$result = $repo->found();
+		$repo->set_use_default_args( true );
+
+		return 0 === $result;
+	}
+
+	/**
+	 * Returns the arguments to query the items.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	private function get_arguments(): array {
+		return array_merge( $this->get_args(), $this->get_instance_args() );
+	}
+
+	/**
+	 * Returns the instance arguments.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	abstract protected function get_instance_args(): array;
+
+	/**
+	 * Returns the arguments to query the items.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	private function get_args(): array {
+		$args = [
+			'orderby' => tec_get_request_var( 'orderby', '' ) ? $this->get_orderby() : 'created',
+			'order'   => tec_get_request_var( 'order', '' ) ? $this->get_order() : 'DESC',
+			'term'    => tec_get_request_var( 's', '' ),
+		];
+
+		return array_filter( $args );
+	}
+
+	/**
+	 * List of sortable columns.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	public function get_sortable_columns() {
+		$column_slugs = array_keys( $this->get_columns() );
+
+		$column_slugs = array_diff( $column_slugs, [ 'cb' ] );
+
+		/**
+		 * Filters the list of sortable columns for the custom table.
+		 *
+		 * @since TBD
+		 *
+		 * @param array $columns List of columns that can be sorted.
+		 */
+		return (array) apply_filters(
+			'tec_common_custom_table_sortable_columns',
+			array_combine( $column_slugs, $column_slugs )
+		);
+	}
+
+	/**
+	 * Handles the checkbox column output.
+	 *
+	 * @since TBD
+	 *
+	 * @param Model $item The current object.
+	 */
+	public function column_cb( $item ): void {
+		$show = current_user_can( 'manage_options' );
+
+		if ( ! $show ) {
+			return;
+		}
+
+		?>
+		<input id="cb-select-<?php echo esc_attr( $item->getPrimaryValue() ); ?>; ?>" type="checkbox" name="item_id[]" value="<?php echo esc_attr( $item->getPrimaryValue() ); ?>" />
+		<label for="cb-select-<?php echo esc_attr( $item->getPrimaryValue() ); ?>; ?>">
```

Comment:
This (and the line above) has an extra PHP closing tag that will mess up the attribute.

```suggestion
		<label for="cb-select-<?php echo esc_attr( $item->getPrimaryValue() ); ?>">
```

Reply:
Also, will the result of `$item->getPrimaryValue()` ever include spaces? Would we want to ensure the value we get is ready for inclusion as an attribute beyond `esc_attr()`?